### ISSUE TITLE
Is it possible to load in Materials serially instead of using CompletableFuture class?

### ISSUE THREAD

**Issue by ChronoAndross** [google-ar/sceneform-android-sdk#123]
Hey sceneform dev team,

I need your help with an issue I'm having. I'm using the RenderableDefinition API and in order to create an object, I need to create a RenderableDefinition.Submesh. In order to create a RenderableDefinition.Submesh without the application crashing, I need to make a material. Right now, the only way to create a material is by using the MaterialFactory class, which provides a material on a background thread. I tried using the .get function of the MaterialFactory return value, but I learned quickly I can't do this because this just blocks the thread forever since the CompletableFuture never finishes. I would like to create a material on the spot since I am working with JNI, and it's a whole lot of extra work to load the materials using the CompletableFuture API after I import my custom geometry. However, I've looked at the API, and it doesn't seem like there is a way to do this. Other Scenegraph APIs can usually create materials on the spot and block the thread until it's finished, and I'm wondering if it's possible to do the same here.

Thanks for your help!

**Comment by malik-at-work** [comment]
Sceneform asset loading is multithreaded and asynchronous.  It's expected that API calls are made from the main thread.  The last stage of the CompletableFuture returns the value to you and executes on the main thread by default.  You can use the CompletableFuture API  to put this on any thread with a message loop pretty easily via one of the Async functions.  Calling CompletableFuture.get() isn't advised unless isDone() is true because it will block.

In summary, if there is an Android message loop on your thread you can shift the results of any work over there pretty easily..If you don't have a message loop I would create a thread for this.
